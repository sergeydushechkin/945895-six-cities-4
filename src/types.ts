enum CardType  {
  MAIN = `MAIN`,
  PROPERTY = `PROPERTY`,
  FAVORITES = `FAVORITES`,
}

enum Type {
  apartment = `apartment`,
  room = `room`,
  house = `house`,
  hotel = `hotel`,
}

enum SortTypes {
  POPULAR = `popular`,
  PRICE_LOW_HIGH = `priceLowHigh`,
  PRICE_HIGH_LOW = `priceHighLow`,
  TOP_RATED_FIRST = `topRatedFirst`
}

enum FavoritesImageSize {
  WIDTH = `150`,
  HEIGHT = `110`,
};

enum OthersImageSize {
  WIDTH = `260`,
  HEIGHT = `200`,
}

interface Offer {
  city: {
    name: string,
    coordinates: [number, number],
    zoom: number,
  }
  id: number,
  title: string,
  previewImage: string,
  price: number,
  type: Type,
  rating: number,
  isPremium: boolean,
  isFavorite: boolean,
  pictures: Array<string>,
  bedrooms: number,
  description: string,
  guests: number,
  features: Array<string>,
  host: {
    avatarUrl: string,
    id: number,
    isPro: boolean,
    name: string,
  },
  location: {
    coordinates: [number, number],
    zoom: number,
  },
}

interface Comment {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: {
    avatarUrl: string,
    id: number,
    isPro: boolean,
    name: string,
  },
}

interface AuthInfo {
  avatarUrl: string,
  email: string,
  id: number,
  isPro: boolean,
  name: string,
}

export {CardType, SortTypes, Type, Offer, Comment, AuthInfo, FavoritesImageSize, OthersImageSize};
