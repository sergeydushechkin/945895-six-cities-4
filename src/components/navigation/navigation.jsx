import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

import {AuthorizationStatus} from "../../reducer/user/user.js";
import {getAuthStatus, getAuthInfo} from "../../reducer/user/selectors.js";

import {AppRoute} from "../../const.js";

const Navigation = (props) => {
  const {authStatus, authInfo} = props;

  return (
    <nav className="header__nav">
      <ul className="header__nav-list">
        <li className="header__nav-item user">
          <Link
            className="header__nav-link header__nav-link--profile"
            to={authStatus === AuthorizationStatus.AUTH ? AppRoute.FAVORITES : AppRoute.LOGIN}
          >
            <div className="header__avatar-wrapper user__avatar-wrapper">
            </div>
            {
              authStatus === AuthorizationStatus.AUTH
                ? <span className="header__user-name user__name">{authInfo.email}</span>
                : <span className="header__login">Sign in</span>
            }
          </Link>
        </li>
      </ul>
    </nav>
  );
};

Navigation.propTypes = {
  authStatus: PropTypes.oneOf([AuthorizationStatus.AUTH, AuthorizationStatus.NO_AUTH]).isRequired,
  authInfo: PropTypes.shape({
    avatarUrl: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    id: PropTypes.any,
    isPro: PropTypes.bool,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

const mapStateToProps = (state) => {
  return {
    authStatus: getAuthStatus(state),
    authInfo: getAuthInfo(state),
  };
};

export {Navigation};
export default connect(mapStateToProps)(Navigation);
