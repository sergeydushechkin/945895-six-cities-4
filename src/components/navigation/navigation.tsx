import * as React from "react";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

import {AuthorizationStatus} from "../../reducer/user/user";
import {getAuthStatus, getAuthInfo} from "../../reducer/user/selectors";

import {AppRoute} from "../../const";
import {AuthInfo} from "../../types";

interface Props {
  authStatus: AuthorizationStatus;
  authInfo: AuthInfo;
}

const Navigation: React.FunctionComponent<Props> = (props: Props) => {
  const {authStatus, authInfo} = props;

  return (
    <nav className="header__nav">
      <ul className="header__nav-list">
        <li className="header__nav-item user">
          <Link
            className="header__nav-link header__nav-link--profile"
            to={authStatus === AuthorizationStatus.AUTH ? AppRoute.FAVORITES : AppRoute.LOGIN}
          >
            <div className="header__avatar-wrapper user__avatar-wrapper">
            </div>
            {
              authStatus === AuthorizationStatus.AUTH
                ? <span className="header__user-name user__name">{authInfo.email}</span>
                : <span className="header__login">Sign in</span>
            }
          </Link>
        </li>
      </ul>
    </nav>
  );
};

const mapStateToProps = (state) => {
  return {
    authStatus: getAuthStatus(state),
    authInfo: getAuthInfo(state),
  };
};

export {Navigation};
export default connect(mapStateToProps)(Navigation);
