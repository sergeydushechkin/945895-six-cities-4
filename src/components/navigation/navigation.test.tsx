import * as React from "react";
import * as renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import {AuthInfo} from "../../types";
import history from "../../history";
import {Navigation} from "./navigation";

const authInfo: AuthInfo = {
  avatarUrl: `img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  isPro: false,
  name: `Oliver.conner`
};

it(`Render Navigation AUTH`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Navigation
            authStatus={`AUTH`}
            authInfo={authInfo}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Navigation NO AUTH`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Navigation
            authStatus={`NO_AUTH`}
            authInfo={authInfo}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
