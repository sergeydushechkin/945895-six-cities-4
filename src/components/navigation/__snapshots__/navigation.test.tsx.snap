// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navigation AUTH 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="/favorites"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__user-name user__name"
        >
          Oliver.conner@gmail.com
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Render Navigation NO AUTH 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="/login"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__login"
        >
          Sign in
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
